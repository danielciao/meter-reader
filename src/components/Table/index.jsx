import styled from 'styled-components';
import { rem } from 'polished';

export default styled.table`
  border-collapse: collapse;
  width: 100%;
`;

export const TableHeader = styled.thead`
  th:not(:last-of-type) {
    border-right: solid 1px white;
  }
`;

export const TableBody = styled.tbody``;

export const TableCaption = styled.caption`
  caption-side: bottom;
  padding: ${rem(10)} ${rem(20)};
  text-align: left;
  font-style: italic;
`;

export const TableRow = styled.tr`
  &:nth-child(even) {
    background: #cbd5df;
  }
`;

export const CellHeader = styled.th`
  color: white;
  background: #00325c;
  font-weight: 600;
  padding: ${rem(10)} ${rem(20)};
`;

export const CellData = styled.td`
  padding: ${rem(10)} ${rem(20)};
`;
