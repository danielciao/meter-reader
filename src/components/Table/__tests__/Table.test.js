import React from 'react';
import renderer from 'react-test-renderer';

import Table, { TableHeader, TableBody, TableCaption, TableRow, CellHeader, CellData } from '../index';

describe('Table', () => {
  it('should render Table correctly', () => {
    const result = renderer
      .create(
        <Table>
          <TableHeader>
            <TableRow>
              <CellHeader>Account Name</CellHeader>
              <CellHeader>Account Type</CellHeader>
              <CellHeader>Balance</CellHeader>
            </TableRow>
          </TableHeader>
          <TableBody>
            <TableRow>
              <CellData>Monthly Budget</CellData>
              <CellData>Current</CellData>
              <CellData>350</CellData>
            </TableRow>
            <TableRow>
              <CellData>Holiday Fund</CellData>
              <CellData>Savings</CellData>
              <CellData>1800</CellData>
            </TableRow>
          </TableBody>
          <TableCaption>* Your account summary.</TableCaption>
        </Table>
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
