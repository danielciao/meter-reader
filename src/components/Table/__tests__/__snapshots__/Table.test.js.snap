// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render Table correctly 1`] = `
<table
  className="sc-bdVaJa hPLTEz"
>
  <thead
    className="sc-bwzfXH hphvth"
  >
    <tr
      className="sc-ifAKCX hNBzmR"
    >
      <th
        className="sc-EHOje fzCMNg"
      >
        Account Name
      </th>
      <th
        className="sc-EHOje fzCMNg"
      >
        Account Type
      </th>
      <th
        className="sc-EHOje fzCMNg"
      >
        Balance
      </th>
    </tr>
  </thead>
  <tbody
    className="sc-htpNat KtFsv"
  >
    <tr
      className="sc-ifAKCX hNBzmR"
    >
      <td
        className="sc-bZQynM laqJsW"
      >
        Monthly Budget
      </td>
      <td
        className="sc-bZQynM laqJsW"
      >
        Current
      </td>
      <td
        className="sc-bZQynM laqJsW"
      >
        350
      </td>
    </tr>
    <tr
      className="sc-ifAKCX hNBzmR"
    >
      <td
        className="sc-bZQynM laqJsW"
      >
        Holiday Fund
      </td>
      <td
        className="sc-bZQynM laqJsW"
      >
        Savings
      </td>
      <td
        className="sc-bZQynM laqJsW"
      >
        1800
      </td>
    </tr>
  </tbody>
  <caption
    className="sc-bxivhb jGGQlA"
  >
    * Your account summary.
  </caption>
</table>
`;
